openlibrary/plugins/worksearch/code.py

```diff
@@ -631,7 +631,7 @@ def works_by_author(akey, sort='editions', page=1, rows=100, has_fulltext=False,
     fq = ['author_key:' + akey, 'type:work']
     if has_fulltext:
         fq.append('has_fulltext:true')
-    solr_select = solr_select_url + "?fq=%s&q.op=AND&q=%s&start=%d&rows=%d&fl=%s&wt=json" % ('&fq='.join(fq), q, offset, rows, fl)
+    solr_select = solr_select_url + "?fq=%s&q.op=AND&q=%s&start=%d&rows=%d&fl=%s&wt=json" % ('&fq='.join(fq), web.urlquote(q), offset, rows, fl)
```

Comment:
This solution looks good to me -- the linter is just complaining about the line length.

@nsachin08 can we make the GET parameters a dictionary instead of a string of values and then encode them into the url (in a way that works for both python2+3).

`urllib.parse.urlencode` (python3), `urllib.urlencode(query[, doseq])` (python2)

Reply:
@mekarpeles  I gave a new start . So created a new PR  #4215  with dictionary implemantion. And closing this.